api/src/main/java/org/pmiops/workbench/api/UserMetricsController.java

```diff
@@ -172,6 +172,29 @@ private long getWorkspaceId(String workspaceNamespace, String workspaceId) {
     Workspace dbWorkspace = workspaceService.getRequired(workspaceNamespace, workspaceId);
     return dbWorkspace.getWorkspaceId();
   }
-}
 
+  /**
+   * Utility method to parse a URI from a string that may or may not have spaces in the path.
+   *
+   * @param str The string to convert
+   * @return The converted URI
+   * @throws URISyntaxException The URISyntaxException
+   */
+  private URI convertToURI(String str) throws URISyntaxException {
```

Comment:
Would it be valuable to put this in a utils file rather than as a function within the component? Maybe here: https://github.com/all-of-us/workbench/blob/master/ui/src/app/utils/index.ts

Reply:
I don't think this is a very generic kind of thing - it's a very narrow use case. Looking on the java side of things, we don't really have a great home for this kind of thing. I'd prefer to punt on this.